src/skimage/registration/_affine.py

```diff
@@ -0,0 +1,551 @@
+"""Affine registration"""
+
+import warnings
+
+from functools import partial
+from itertools import product, combinations_with_replacement, combinations
+
+import numpy as np
+from scipy import ndimage as ndi
+from scipy.optimize import minimize
+
+from skimage.transform.pyramids import pyramid_gaussian
+from skimage.metrics import normalized_mutual_information
+
+from math import log, floor, cos, sin, pow
+
+
+def target_registration_error(shape, matrix):
+    """
+    Compute the displacement norm of the transform at at each pixel.
```

Comment:
```suggestion
    """Compute the displacement norm of the transform at each pixel.
```
Reference: https://peps.python.org/pep-0257/#multi-line-docstrings